client/src/handlers/handlers.ts

```diff
@@ -14,6 +19,211 @@ export function makeHandleRunJob (
   }
 }
 
+/**
+ * returns an active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
+ *
+ * @return vscode.Terminal
+*/
+function ensureFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (const element of activeTerminals) {
+        if(element.name.substring(0, 4) == `flix`)
+            return element
+    }
+    const terminal = vscode.window.createTerminal(`flix-`+countTerminals.toString());
+    countTerminals+=1 //creating a new terminal since no active flix terminals available.
+    return terminal
+}
+
+
+/**
+ * returns an new active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix` and returns it.
+ *
+ * If there are already `n` active terminals exist with prefix name `flix`, it creates a new terminal with name `flix n+1`
+ *
+ * @return vscode.Terminal
+*/
+function ensureNewFlixTerminal() {
+    const terminal = vscode.window.createTerminal(`flix-`+countTerminals.toString())
+    countTerminals+=1
+    return terminal
+}
+
+/**
+ * takes a string and a terminal and passes that string to the terminal.
+ *
+ * @param cmd string (a terminal command) to pass to the terminal.
+ *
+ * @param terminal vscode.Terminal 
+ *
+ * @return void
+*/
+function passCommandToTerminal(cmd:string, terminal: vscode.Terminal) {
+    terminal.show();
+    terminal.sendText(cmd);
+}
+
+/**
+ * Opens an input box to ask the user for input.
+ * uses the `vscode.window.showInputBox` function with custom `prompt` and `placeHolder` and value of `ignoreFocusOut` to be `true`.
+ *
+ *
+ * @return A promise that resolves to a string the user provided or to `undefined` in case of dismissal.
+*/
+async function takeInputFromUser() {
+    const input = await vscode.window.showInputBox({
+        prompt: "Enter arguments separated by spaces",
+        placeHolder: "arg0 arg1 arg2 ...",
+		ignoreFocusOut: true
+    })
+    return input
+}
+
+/**
+ * combines the paths of all flix files present in the current directory of vscode window.
+ * 
+ * gets an array of vscode.Uri for all flix files using `vscode.workspace.findFiles` function
+ *
+ * @return string of format "\<path_to_first_file\>" "\<path_to_second_file\>" ..........
+*/
+async function getFiles() {
+    const files = await vscode.workspace.findFiles(FLIX_GLOB_PATTERN)
+    let paths = files.map(x => x.path)
+    let cmd = quote(paths)
+    return cmd
+}
+
+/**
+ * sends a java command to compile and run flix program of vscode window to the terminal.
+ *
+ * @param terminal vscode.Terminal to receive the command. 
+ *
+ * @return void
+*/
+async function passArgs(terminal:vscode.Terminal, flixFilename: string) {
+
+    let cmd = "java -jar "+ flixFilename + " "
```

Comment:
Refactoring:

Instead of calling quote on `flixFilename` and building a string here, just build one array:

`['java', '-jar',flixFileName]`

and then push the arguments on to the array (if any). 

I.e. the point is to **completely avoid string concatenation**.